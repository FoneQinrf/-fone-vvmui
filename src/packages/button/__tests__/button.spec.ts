/*
 * @Author: Fone`峰
 * @Date: 2021-05-16 10:51:45
 * @LastEditors: Fone`峰
 * @LastEditTime: 2021-05-16 12:02:07
 * @Description: file content
 * @Email: qinrifeng@163.com
 * @Github: https://github.com/FoneQinrf
 */
import { mount } from '@vue/test-utils';
import Button from '../index.vue';
import { nextTick } from 'vue';

describe('Button', () => {
  const wrapper = mount(Button, {
    slots: {
      default: 'button'
    }
  });

  it('should render type correctly', async () => {
    wrapper.setProps({ type: 'default' });
    await nextTick();
    expect(wrapper.html()).toMatchSnapshot();
    wrapper.setProps({ type: 'success' });
    await nextTick();
    expect(wrapper.html()).toMatchSnapshot();
    wrapper.setProps({ type: 'warning' });
    await nextTick();
    expect(wrapper.html()).toMatchSnapshot();
    wrapper.setProps({ type: 'error' });
    await nextTick();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render size correctly', async () => {
    wrapper.setProps({ type: 'default', size: 'large' });
    await nextTick();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render loading and click correctly', async () => {
    wrapper.setProps({ size: '', loading: true });
    await nextTick();
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.find('.vvm-loading-wrp').exists()).toBe(false);
    wrapper.trigger('click')
    expect(wrapper.emitted()).not.toHaveProperty('click');
  });

  it('should render disabled and click correctly', async () => {
    wrapper.setProps({ disabled: true });
    await nextTick();
    expect(wrapper.html()).toMatchSnapshot();
    wrapper.trigger('click');
    expect(wrapper.emitted()).not.toHaveProperty('click');
  });

  it('should click button correctly', async () => {
    wrapper.setProps({ loading: false, disabled: false });
    await nextTick();
    wrapper.trigger('click');
    expect(wrapper.emitted()).toHaveProperty('click');
  });
});
