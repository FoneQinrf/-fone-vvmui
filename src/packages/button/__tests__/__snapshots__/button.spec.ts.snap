// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render disabled and click correctly 1`] = `
"<button class=\\"vvm-button vvm-button--default vvm-button--disabled vvm-button--loading\\" disabled=\\"\\"><i class=\\"vvm-iconfont vvm-icon vvm-ignore iconloading vvm-loading\\" style=\\"font-size: 18px; width: 18px; height: 18px;\\"></i>
  <!-- <span class=\\"vvm-loading-wrp\\"
        ><Icon class=\\"vvm-loading\\" name=\\"iconloading\\"
      /></span> -->加载中..
</button>"
`;

exports[`Button should render loading and click correctly 1`] = `
"<button class=\\"vvm-button vvm-button--default vvm-button--disabled vvm-button--loading\\" disabled=\\"\\"><i class=\\"vvm-iconfont vvm-icon vvm-ignore iconloading vvm-loading\\" style=\\"font-size: 18px; width: 18px; height: 18px;\\"></i>
  <!-- <span class=\\"vvm-loading-wrp\\"
        ><Icon class=\\"vvm-loading\\" name=\\"iconloading\\"
      /></span> -->加载中..
</button>"
`;

exports[`Button should render size correctly 1`] = `
"<button class=\\"vvm-button vvm-button--default vvm-button--large\\">
  <div class=\\"text\\">button</div>
</button>"
`;

exports[`Button should render type correctly 1`] = `
"<button class=\\"vvm-button vvm-button--default vvm-button--normal\\">
  <div class=\\"text\\">button</div>
</button>"
`;

exports[`Button should render type correctly 2`] = `
"<button class=\\"vvm-button vvm-button--success vvm-button--normal\\">
  <div class=\\"text\\">button</div>
</button>"
`;

exports[`Button should render type correctly 3`] = `
"<button class=\\"vvm-button vvm-button--warning vvm-button--normal\\">
  <div class=\\"text\\">button</div>
</button>"
`;

exports[`Button should render type correctly 4`] = `
"<button class=\\"vvm-button vvm-button--error vvm-button--normal\\">
  <div class=\\"text\\">button</div>
</button>"
`;
